### ISSUE TITLE
Wrong wording of missing CPU instructions

### ISSUE THREAD

**Issue by Flamefire** [tensorflow/tensorflow#51987]
Have a look at the code at https://github.com/tensorflow/tensorflow/blob/3d3c6db1ca2d50f6f07722cd800144f8f736167c/tensorflow/core/platform/cpu_feature_guard.cc#L150-L157

This outputs e.g. 

> This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.

Judging from the naming `missing_instructions` I suppose there is a "not" missing in that warning.

**Comment by svobora** [comment]
This is correct wording. E.g. if CPU supports AVX2 but tensorflow is built without AVX2 support, AVX2 is in missing_instructions. Close this.

**Comment by Flamefire** [comment]
So a TensorFlow build without avx2 enabled which builds oneDNN alongside its build doesn't use avx2, but the oneDNN part does use it?
Intresting...
Then maybe the wording can be improved?

**Comment by sanatmpa1** [comment]
@Flamefire,

As the wording is correct already, Can you suggest how the wording can be improved ?

**Comment by Flamefire** [comment]
"The TensorFlow binaries were not build with the following CPU instruction sets: <...> To enable them rebuild ...\n Note that oneDNN is used for various performance-critical operations and may dynamically use those CPU instructions sets nonetheless."

Or so... And maybe make sure that oneDNN is actually used/built because I don't see any switch checking if oneDNN support is enabled around that message.

**Comment by sanatmpa1** [comment]
@Flamefire,

Can you submit a PR for updating this [file](https://github.com/tensorflow/tensorflow/blob/3d3c6db1ca2d50f6f07722cd800144f8f736167c/tensorflow/core/platform/cpu_feature_guard.cc#L150-L157) with the change you want to propose? Thanks!
